key.go

```diff
@@ -0,0 +1,69 @@
+package statter
+
+import (
+	"sync"
+	"unsafe"
+)
+
+var keyPool = sync.Pool{
+	New: func() interface{} {
+		return &key{b: make([]byte, 0, 256)}
+	},
+}
+
+type key struct {
+	b []byte
+}
+
+func newKey(name string, tags []Tag) *key {
+	k := keyPool.Get().(*key)
+	k.b = k.b[:0]
+
+	k.b = append(k.b, name...)
```

Comment:
What's the point of resetting it then appending to the empty slice rather than just directly replacing it with the slice of bytes that `name` represents?

Reply:
I want to reuse the `key` byte slice which will, thanks to the pool, be longer than the name. If I were to do as you suggest, it would alloc every time it is used as the name slice (string and slice are almost the same) has a `cap` as long as the string, and it still needs to add tags. This way the key bytes in the pool will grow to an average key length and very rarely need to expand.